pageserver/src/tenant/timeline/layer_manager.rs

```diff
@@ -69,21 +69,14 @@ impl LayerManager {
     pub(crate) async fn get_layer_for_write(
         &mut self,
         lsn: Lsn,
-        last_record_lsn: Lsn,
         conf: &'static PageServerConf,
         timeline_id: TimelineId,
         tenant_shard_id: TenantShardId,
+        gate_guard: utils::sync::gate::GateGuard,
         ctx: &RequestContext,
     ) -> Result<Arc<InMemoryLayer>> {
         ensure!(lsn.is_aligned());
 
-        ensure!(
-            lsn > last_record_lsn,
-            "cannot modify relation after advancing last_record_lsn (incoming_lsn={}, last_record_lsn={})",
-            lsn,
-            last_record_lsn,
-        );
-
```

Comment:
I just moved this back to timeline (only caller) because that's where the last_record_lsn is to avoid a `#[allow(clippy::too_many_args)]`.

Reply:
I think the ensure makes more sense here than in the caller. Just slap on the allow(too_many_arguments), why not?

Reply:
It is irresponsible to spray allow lints everywhere without consideration. Timeline has the last_record_lsn, so I think it should enforce this.. But you don't think that is a good enough barrier because Timeline::layers is pub(crate) or pub? I kind of agree.